pkg/api/defaults.go

```diff
@@ -217,6 +217,10 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpdate bool) {
 				// TODO make EnableAggregatedAPIs a pointer to bool so that a user can opt out of it
 				a.OrchestratorProfile.KubernetesConfig.EnableAggregatedAPIs = true
 			}
+		} else if isUpdate && a.OrchestratorProfile.KubernetesConfig.EnableAggregatedAPIs {
+			// Upgrade scenario:
+			// We need to force set EnableAggregatedAPIs to false if RBAC was previously disabled
```

Comment:
"if RBAC was previously disabled" --> where is that logic? 

This seems to always set EnableAggregatedAPIs to `false` if it was true during update which doesn't seem to be the behavior we want.

Reply:
This is in the "else" block of a `if a.OrchestratorProfile.KubernetesConfig.IsRBACEnabled() {` statement. So the brute force override only happens if RBAC is disabled.

Reply:
aahh didn't read the diff correctly